misc-scripts/xref_mapping/XrefParser/RefSeqCoordinateParser.pm

```diff
@@ -23,405 +23,300 @@ use strict;
+    croak "No data found for RefSeq_import, skipping import\n";
   }
 
-  foreach my $chromosome_of (@$chromosomes_of) {
+  # start main loop
+  foreach my $chromosome_of (@{$chromosomes_of}) {
     my $chr_name = $chromosome_of->seq_region_name();
-    my $genes_of = $chromosome_of->get_all_Genes($logic_name, undef, 1);
+    my $genes_of = $chromosome_of->get_all_Genes('refseq_import', undef, 1);
 
-    while (my $gene_of = shift @$genes_of) {
+    foreach my $gene_of (@{$genes_of}) {
       my $transcripts_of = $gene_of->get_all_Transcripts();
 
-# Create a range registry for all the exons of the refseq transcript
-      foreach my $transcript_of (sort { $a->start() <=> $b->start() } @$transcripts_of) {
-	my ($id, $tl_id);
-	# We're moving to RefSeq accessions being stored as xrefs rather than
-	# stable ids. But we also need to maintain backwards compatbility.
-	# If it's the new kind, where there's a display_xref use that,
-	# otherwise fall back to using the stable_id. But also check if we
-	# have neither, then skip the record.
-	if (defined $transcript_of->display_xref ) {
-	  $id = $transcript_of->display_xref->display_id;
-	} elsif (defined $transcript_of->stable_id) {
-	  $id = $transcript_of->stable_id;
-	} else {
-	  # Skip non conventional accessions
-	  next;
-	}
-        if ($id !~ /^[NXMR]{2}_[0-9]+/)  { next; }
-        my %transcript_result;
-        my %tl_transcript_result;
-        if (!defined $id) { next; }
+      # Create a range registry for all the exons of the refseq transcript
+      foreach my $transcript_of (sort { $a->start <=> $b->start } @{$transcripts_of}) {
+        my $id;
+        # We're moving to RefSeq accessions being stored as xrefs rather than
+        # stable ids. But we also need to maintain backwards compatbility.
+        # If it's the new kind, where there's a display_xref use that,
+        # otherwise fall back to using the stable_id. But also check if we
+        # have neither, then skip the record.
+        if (defined $transcript_of->display_xref ) {
+          $id = $transcript_of->display_xref->display_id;
+        } elsif (defined $transcript_of->stable_id) {
+          $id = $transcript_of->stable_id;
+        }
+        # Skip non conventional accessions
+        unless ( $id =~ /\A[NXMR]{2}_\d+\.?\d+\z/xms ) {
+          next;
+        }
+
+        my $transcript_result;
+        my $tl_transcript_result;
+
         my $exons_of = $transcript_of->get_all_Exons();
         my $rr1 = Bio::EnsEMBL::Mapper::RangeRegistry->new();
         my $tl_exons_of = $transcript_of->get_all_translateable_Exons();
         my $rr3 = Bio::EnsEMBL::Mapper::RangeRegistry->new();
 
-        foreach my $exon_of (@$exons_of) {
-          my $start_of = $exon_of->seq_region_start();
-          my $end_of = $exon_of->seq_region_end();
-          $rr1->check_and_register( 'exon', $start_of, $end_of );
-        }
+        # register $exons_of on $rr1
```

Comment:
Perhaps rr1,2,3,4 and could be named more functionally, now you've cleaned up the mess? I reckon a longer variable name could be forgiven here where there is so much room for confusion